File: 148.cpp
#include<iostream>
#include<algorithm>
using namespace std;
typedef long long ll;
#define noch int t;cin>>t;
#define yes cout<<"Yes"<<endl;
#define no cout<<"No"<<endl;
ll fact(ll n){
    ll p=1;
    for(ll i=1;i<=n;i++){
        p*=i;
    }
    return p;
}
int main()
{
ll a,b;
cin>>a>>b;
ll ans=fact(min(a,b));
cout<<ans<<endl;
return 0;
}

File: 148.py
def fact(n):
    p = 1
    for i in range(1, n + 1):
        p *= i
    return p

def gcd_of_factorials(a, b):
    smaller = min(a, b)
    return fact(smaller)

def main():
    a, b = map(int, input("Enter two numbers (a and b): ").split())
    ans = gcd_of_factorials(a, b)
    print(ans)

if __name__ == "__main__":
    main()
